### DISCUSSION TITLE
Usage of ETL and Tricore Architecture

### DISCUSSION THREAD

**Discussion by ef12** [ETLCPP/etl#713] [Q&A]
Hello everyone,

I am currently involved in developing embedded software for the Aurix TC38xx MCU, which utilizes the tricore architecture. This particular architecture is notorious for being unforgiving when it comes to memory misalignment issues.

I would like to inquire about the status of the ETL in relation to this problem. How effectively does the ETL address memory misalignment issues in the context of the Aurix TC38xx MCU?

Thank you.

**Comment by MahsaBakhtiari** [comment]
Hey there, I totally understand how challenging it can be, especially when dealing with memory misalignment issues on the tricore architecture. But don't worry, you're not alone in this!
Regarding the ETL (Embedded Template Library), it's actually known for doing a pretty good job at addressing memory misalignment problems specifically for the Aurix TC38xx MCU. It's designed to provide optimized data structures and algorithms that work well with the tricore architecture.

**Reply by jwellbelove** [reply]
Thanks for the input Mahsa. I've not had experience with the Aurix TC38xx MCU myself.
